### PR TITLE
Feature/cab 2544

### PR DESCRIPTION
**Pull request by maximede**

### CODE REVIEW

File: src/app/routing/config-resolver.service.ts [hunk 1]
```diff
@@ -0,0 +1,38 @@
...
+  }
+
+  resolve(route: ActivatedRouteSnapshot, state: RouterStateSnapshot): Promise<Config> {
+
+    let tenant: string = route.params.tenant;
+
+    if (!tenant) {
+      // Try to find a tenant param in the parent routes
+      while (!tenant && route.parent) {
+        tenant = route.parent.params.tenant;
```
Comment (agagne) [inline src/app/routing/config-resolver.service.ts]: is it possible that a tenant might never be found?

Comment (maximede) [inline src/app/routing/config-resolver.service.ts]: yes, and it should redirect the user to `/` if that's the case

File: src/app/pages/tenant/tenant.component.spec.ts [hunk 2]
```diff
@@ -0,0 +1,82 @@
...
+        {provide: ActivatedRoute, useValue: activatedRoute},
+        Title,
+      ],
+      schemas: [NO_ERRORS_SCHEMA]
+    })
+      .compileComponents();
+  }));
+
+
+  describe('when component is created and a correct config is sent', () => {
```
Comment (huiaic) [inline src/app/pages/tenant/tenant.component.spec.ts]: This description doesn't quite flow with the specs'. How about just "Component", or add "... then the component" at the end.

File: src/app/widgets/header/header.component.ts [hunk 3]
```diff
@@ -0,0 +1,31 @@
...
+import { User } from '../../user/user';
+
+@Component({
+  selector: 'app-header',
+  templateUrl: './header.component.html',
+  styleUrls: ['./header.component.css']
+})
+export class HeaderComponent implements OnInit {
+  tenant: string;
+  promiseUser: Promise<User>;
```
Comment (agagne) [inline src/app/widgets/header/header.component.ts]: probably should rename to just User, rather than encoding that it is enveloped in promise in the name

**Pull request merged**